volume/volume_windows.go

```diff
@@ -0,0 +1,163 @@
+package volume
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	derr "github.com/docker/docker/errors"
+)
+
+// read-write modes
+var rwModes = map[string]bool{
+	"rw": true,
+}
+
+// read-only modes
+var roModes = map[string]bool{
+	"ro": true,
+}
+
+const (
+	// Spec should be in the format [source:]destination[:mode]
+	//
+	// Examples: c:\foo bar:d:rw
+	//           c:\foo:d:\bar
+	//           myname:d:
+	//           d:\
+	//
+	// Explanation of this regex! Thanks @thaJeztah on IRC and gist for help. See
+	// https://gist.github.com/thaJeztah/6185659e4978789fb2b2. A good place to
+	// test is https://regex-golang.appspot.com/assets/html/index.html
+	//
+	// Useful link for referencing named capturing groups:
+	// http://stackoverflow.com/questions/20750843/using-named-matches-from-go-regex
+	//
+	// There are three match groups: source, destination and mode.
+	//
+
+	// RXHostDir is the first option of a source
+	RXHostDir = `[a-zA-Z]:\\(?:[^\\/:*?"<>|\r\n]+\\?)*`
+	// RXName is the second option of a source
+	RXName = `[a-zA-Z0-9][a-zA-Z0-9_.-]+`
```

Comment:
> Gosh, do you have children?

2 adolescents ;)

That link is super informative, thanks. Let's implement exactly those restrictions, no more, no less.

> until there is direct end-user feedback that this is not workable - have you any data to say that this is the case?

Indeed, I can point you to several issues where people are confused because we are too restrictive with our name policies. I can also point you to a couple of discussions where not having these restrictions in volume names have been a blessing from the user point of view.

> Note also that directory names are case insensitive on NTFS, so it's always lower-cased on Windows anyway - the volume name "John" will equate to "john" during processing - there is no way we can avoid that on Windows.

Cool, again, it's not our problem if people don't understand how NTFS fundamentally works.